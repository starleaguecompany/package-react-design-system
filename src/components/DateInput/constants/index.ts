export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
