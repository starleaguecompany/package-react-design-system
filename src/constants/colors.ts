export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
