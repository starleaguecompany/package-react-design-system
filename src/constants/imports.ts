export const reactDSImportPath = '@starleaguecompany/react-design-system';
