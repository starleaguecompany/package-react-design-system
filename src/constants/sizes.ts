export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
