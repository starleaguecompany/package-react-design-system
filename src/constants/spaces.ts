export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
