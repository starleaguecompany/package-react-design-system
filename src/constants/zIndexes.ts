export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
