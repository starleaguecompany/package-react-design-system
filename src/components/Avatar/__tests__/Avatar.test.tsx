import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { CONTAINER_SIZES } from '../../../constants/sizes';
import { Avatar } from '../index';

describe('Avatar', () => {
  test('should render correctly with no props', () => {
    render(<Avatar />);

    expect(screen.getByTestId('Avatar')).toMatchSnapshot();
    expect(screen.getByTestId('Avatar')).toHaveClass('avatar size-44', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Avatar id="test-id" className="test-class" />);

    expect(screen.getByTestId('Avatar')).toMatchSnapshot();
    expect(screen.getByTestId('Avatar')).toHaveClass('test-class');
    expect(screen.getByTestId('Avatar')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with src prop', () => {
    render(<Avatar src="test" />);

    expect(screen.getByTestId('Avatar')).toMatchSnapshot();
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveClass('image');
    expect(screen.getByRole('img')).toHaveAttribute('src', 'test');
    expect(screen.getByRole('img')).toHaveAttribute('loading', 'lazy');
  });

  test('should render correctly with text', () => {
    render(<Avatar>test</Avatar>);

    expect(screen.getByTestId('Avatar')).toMatchSnapshot();
    expect(screen.getByTestId('Avatar')).toHaveTextContent('test');
    expect(screen.getByText('test')).toBeVisible();
  });

  test('should render correctly with size prop', () => {
    render(<Avatar size={CONTAINER_SIZES.S60}>test</Avatar>);

    expect(screen.getByTestId('Avatar')).toMatchSnapshot();
    expect(screen.getByTestId('Avatar')).toHaveClass('size-60');
  });
});
