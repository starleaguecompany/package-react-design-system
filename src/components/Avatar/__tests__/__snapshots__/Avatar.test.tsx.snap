// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly with attributes 1`] = `
<div
  class="test-class avatar size-44"
  data-qa="Avatar"
  id="test-id"
>
  <div
    class="size-16 strong uppercase"
  />
</div>
`;

exports[`Avatar should render correctly with no props 1`] = `
<div
  class="avatar size-44"
  data-qa="Avatar"
>
  <div
    class="size-16 strong uppercase"
  />
</div>
`;

exports[`Avatar should render correctly with size prop 1`] = `
<div
  class="avatar size-60"
  data-qa="Avatar"
>
  <div
    class="size-20 strong uppercase"
  >
    test
  </div>
</div>
`;

exports[`Avatar should render correctly with src prop 1`] = `
<div
  class="avatar size-44"
  data-qa="Avatar"
>
  <img
    alt="avatar"
    class="image"
    loading="lazy"
    src="test"
  />
</div>
`;

exports[`Avatar should render correctly with text 1`] = `
<div
  class="avatar size-44"
  data-qa="Avatar"
>
  <div
    class="size-16 strong uppercase"
  >
    test
  </div>
</div>
`;
